var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -93,7 +93,9 @@ class PyNumpy(PythonPackage):
     depends_on("python@3.6:3.10", type=("build", "link", "run"), when="@1.19")
     depends_on("python@3.7:3.10", type=("build", "link", "run"), when="@1.20:1.21")
     depends_on("python@3.8:", type=("build", "link", "run"), when="@1.22:")
-    depends_on("py-setuptools@:59", type=("build", "run"))
+    # https://github.com/spack/spack/pull/32078
+    depends_on("py-setuptools@:63", type=("build", "run"))
+    depends_on("py-setuptools@:59", when="@:1.22.1", type=("build", "run"))
```

Comment:
Setuptools 59 is fine, 60–63 raise deprecation warnings, and 64 fails entirely. Unfortunately, Spack cannot yet resolve build deps separately, and each DAG must contain a single version of each package. Since shapely now requires newer setuptools, I've relaxed this dep to support this. Once Spack can resolve deps separately, we can go back to pinning this.